routes/blogs.js

```diff
@@ -0,0 +1,126 @@
+const express = require('express');
+const router = express.Router();
+const log = require('../js/log');
+const constants = require('../helpers/constants');
+const mongoose = require('mongoose');
+const db = require('../js/db');
+const Blog = require('../models/blog');
+const helper = require('../helpers/helper');
+const messages = require('../helpers/messages');
+
+router.get('/', (req, res, next) => {
+  Blog.find()
+    .then(blogs => {
+      res.json(blogs);
+    })
+    .catch(err => {
+      helper.prepareError(err, constants.SERVER_ERROR, messages.ERROR_GET);
+      next(err);
+    })
+});
+
+router.post('/', (req, res, next) => {
+  const newBlog = Object.assign(new Blog(), {
+    title: req.body.title,
+    author: req.decoded.name,
+    body: req.body.body,
+    date: new Date(new Date().toISOString()),
+  });
+
+  var error = newBlog.validateSync();
+  if (error && error.errors) {
+    res.sendStatus(constants.SERVER_ERROR);
+  }
+
+  Blog.create(newBlog)
+    .then(blog => {
+      res.json({ message: "New blog Created!" });
+    })
+    .catch(err => {
+      helper.prepareError(err, constants.SERVER_ERROR, messages.ERROR_CREATE);
+      next(err);
+    });
+});
+
+router.get('/:id', (req, res, next) => {
+  const { id } = req.params;
+  if (mongoose.Types.ObjectId.isValid(id)) {
```

Comment:
Maybe it would be cool to define middleware where you can do this validation?

Reply:
done